src/Chapter3.hs

```diff
@@ -910,6 +1008,21 @@ Implement instances of "Append" for the following types:
 class Append a where
     append :: a -> a -> a
 
+newtype Gold = Gold Int
+
+instance Append Gold where
+  append :: Gold -> Gold -> Gold
+  append (Gold x) (Gold y) = Gold (x + y)
+
+instance Append [a] where
+  append :: [a] -> [a] -> [a]
+  append x y = x ++ y
+
+instance Append a => Append (Maybe a) where
+  append :: Maybe a -> Maybe a -> Maybe a
+  append Nothing _ = Nothing
+  append _ Nothing = Nothing
```

Comment:
While this is valid implementation and nothing in the task description says differently, in reality, it would be more beneficial is this function returned `Just` when at least one argument is `Just`.

The `Maybe` instance for the real similar typeclass in Haskell is implemented by returning `Just`s 🙂